provider/repodata/data/json/props/names/to/short/ccc@1/und.json

```diff
@@ -0,0 +1,62 @@
+{
+  "map": {
```

Comment:
Observation: except for CCC, all these properties have contiguous discriminants. It would be smaller and faster to store most properties as `VarZeroVec<str>` and fall back to `ZeroMap<u16, str>` only when needed.

This could also be done later as an optimization in ZeroMap itself (not exactly sure how: perhaps deserializing could set some flag on the ZeroMap that the index list is contiguous).

Reply:
So some kind of enum between the two?  Makes sense, could do it in a bit.

Reply:
Yeah, I'm only worried about perf impact of the extra branch. Would be nice if we could get rid of it in `as_borrowed` somehow.

Reply:
Or we _could_ store every property as `VarZeroVec<str>`, with empty strings as sentinels, hurting the data size for CCC but improving it for everyone else

Reply:
I was thinking about that, but wasn't sure if that would work for all properties now and future.

Reply:
Though it is tempting because it would avoid the future issue where a deliberately allocated blank (eg a script) flips the representation.

Reply:
I experimented a bit, I like the enum version. One thing I've done is future proofed it by making it so that:

 - linear storage is _allowed_ to have gaps, even though we don't use it right now
 - datagen only stores CCC as a map right now but could freely make this choice for other entries
 - datagen currently errors if it sees any non-ccc properties with gaps, which means that if a gap is introduced in the future, we can patch datagen to either allow it in linear mode, or move it to map mode, on a per-case basis